pipewire/qubes-pw-module.c

```diff
@@ -497,17 +496,36 @@ static int process_control_commands(struct impl *impl)
         return -EPIPE;
     }
 
-    if (new_state != playback_stream->last_state) {
+    while (new_state != playback_stream->last_state) {
         uint32_t cmd = new_state ? QUBES_PA_SINK_UNCORK_CMD : QUBES_PA_SINK_CORK_CMD;
 
+        if (cmd == QUBES_PA_SINK_CORK_CMD) {
+           if (!impl->cork_enable) {
+               playback_stream->last_state = new_state;
+               break;
+           }
+
+           if (playback_stream->vchan) {
+               int space = libvchan_buffer_space(playback_stream->vchan);
+               if (space >= 0 && (size_t)space != playback_stream->buffer_size) {
+                   spa_assert_se((size_t)space < playback_stream->buffer_size);
+                   pw_log_debug("will cork stream when vchan is drained (currently %zu bytes left)",
```

Comment:
How cork cmd will be sent? I can't see it retrying later.

Reply:
`process_control_commands()` is called whenever activity happens on either vchan, unless that vchan is closed.